Watchman.Integrations/MongoDB/Session.cs

```diff
@@ -57,6 +58,11 @@ public Session(IMongoDatabase database)
             await this.GetCollection<T>().DeleteOneAsync(x => x.Id == entity.Id);
         }
 
+        public async Task DeleteAsync<T>(Expression<Func<T, bool>> filter) where T : Entity
+        {
+            await this.GetCollection<T>().DeleteManyAsync(filter);
```

Comment:
zastanawiam sie czy rzeczywiście chcemy usuwać dane z bazy 
myśle że dużo lepszym sposobem będzie zmiana tego i `await this.GetCollection<T>().DeleteOneAsync(x => x.Id == entity.Id);`
na zmienianie propertiesa IsDeleted na true w encjach
i przy pobieraniu - filtrowanie na poziomie Session po IsDeleted 

jak jest dużo kodu to łatwo o drobne błędy których ktoś nie zauważy
i wychodzą dopiero po długim czasie
albo ktoś źle zrozumie zadanie, inna osoba to zatwierdzi bo nie zauważy
i mamy problem (kilka razy już tak było)

myśle że co jakiś czas powinien działać w tle proces backupujący baze danych i czyszczący ją z usuniętych już rekordów (IsDeleted)
ale dopiero po upewnieniu sie że backup udał sie w pełni prawidłowo - nie zginęły tam przypadkiem żadne dane itd

@gagyn co myślisz?

Reply:
można dodać IsDeleted do Entity i tylko to zmieniać